maven-artifact/src/main/java/org/apache/maven/artifact/ArtifactUtils.java

```diff
@@ -56,15 +56,20 @@ public static String toSnapshotVersion( String version )
     {
         Validate.notBlank( version, "version can neither be null, empty nor blank" );
 
-        Matcher m = Artifact.VERSION_FILE_PATTERN.matcher( version );
-        if ( m.matches() )
+        int lastDash = version.lastIndexOf( '-' );
```

Comment:
That's a hyphen, not a dash

Reply:
That's true, you want me to change the variable name ?

Reply:
Yes, please.